pkg/kubectl/resource_printer.go

```diff
@@ -412,6 +462,14 @@ func (h *HumanReadablePrinter) HandledResources() []string {
 	return keys
 }
 
+func (h *HumanReadablePrinter) FinishPrint(output io.Writer) error {
+	if !h.options.NoHeaders && !h.options.ShowAll && h.hiddenPodsNum > 0 {
+		_, err := fmt.Fprintf(output, "\n *Be aware of %d pod(s) is(are) in terminated state and hidden from the displayed list!\n", h.hiddenPodsNum)
```

Comment:
If this logic is going to be reused for other objects as in #27510, this message needs to be parameterized somehow. Either the caller passes a custom message or `HumanReadablePrinter` takes care of it in one of two ways:

1. it keeps a mapping of object type -> message
2. it uses a more generic message (i.e. does not mention terminated state). It can either take `%d pod(s)` in the current version that you are suggesting and replace it with `%d object(s)` or it can inject the object type as an additional formatting placeholder, i.e. `%d %s(s)`.